homeassistant/components/sonos/speaker.py

```diff
@@ -193,12 +215,18 @@ def setup(self) -> None:
         else:
             self._platforms_ready.update({BINARY_SENSOR_DOMAIN, SENSOR_DOMAIN})
 
+        self.available_alarms, _ = fetch_alarms_for_speaker(self.soco)
+        if bool(self.available_alarms):
+            dispatcher_send(self.hass, SONOS_CREATE_ALARM, self)
+
+        self._platforms_ready.update({SWITCH_DOMAIN})
+
         dispatcher_send(self.hass, SONOS_CREATE_MEDIA_PLAYER, self)
 
     async def async_handle_new_entity(self, entity_type: str) -> None:
         """Listen to new entities to trigger first subscription."""
         self._platforms_ready.add(entity_type)
-        if self._platforms_ready == PLATFORMS:
+        if self._platforms_ready == PLATFORMS and not self._subscriptions:
```

Comment:
This shouldn't be necessary. Perhaps the `_platforms_ready` call above caused the need for this?

Reply:
No its still necessary, because there might be alarms added during runtime. Upon entity creation they will call `async_handle_new_entity ` and will try to resubscribe.
I think there is no way around it. But I think it doesn't do any harm.